servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -1244,6 +1244,8 @@ float RendererSceneRenderRD::screen_space_roughness_limiter_get_limit() const {
 }
 
 TypedArray<Image> RendererSceneRenderRD::bake_render_uv2(RID p_base, const TypedArray<RID> &p_material_overrides, const Size2i &p_image_size) {
+	ERR_FAIL_COND_V_MSG(p_image_size.width <= 0, TypedArray<Image>(), "Image width must be greater than 0.");
+	ERR_FAIL_COND_V_MSG(p_image_size.height <= 0, TypedArray<Image>(), "Image height must be greater than 0.");
```

Comment:
As seen on #67067 it would likely also crash with width or height higher than Vulkan's `maxFramebufferWidth`, which was 16384 on your Nvidia setup.

I don't know what's the max for other APIs and other drivers but we might want to add a safeguard here too.
Maybe this can or should be queried from the RenderingDevice API. @clayjohn 

But hardcoding 16384 for safety is maybe not a bad idea either to keep it simple.

Reply:
Maximum size can be quered with ``limit_get()``, ``LIMIT_MAX_FRAMEBUFFER_HEIGHT``, and ``LIMIT_MAX_FRAMEBUFFER_WIDTH``

That being said, I think the check should be in ``RD::texture_create()`` which is where the crash is happening. We only validate to ensure the size is at least 1, we should also validate to ensure the size is less than the max:
https://github.com/godotengine/godot/blob/0aba659891f5947d9b55ce7c70aa4d7209369717/drivers/vulkan/rendering_device_vulkan.cpp#L1699

Again, ``limit_get()`` can be used, but with ``LIMIT_MAX_TEXTURE_SIZE_*``

Reply:
The largest working texture size is not necessarily `LIMIT_MAX_TEXTURE_SIZE_*` (which comes from Vulkan's `maxImageDimension*`).

According to the spec, any given image type could have a max image size greater than `maxImageDimension*`: 
> [Some combinations of image parameters (format, usage, etc.) may allow support for larger dimensions](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap45.html)
> [Allowed Extent Values Based On Image Type](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap47.html#features-extentperimagetype)

In addition, certain image usage flags use `maxFramebufferWidth/Height`, which may be smaller than `maxImageDimension*`:
> [VUID-VkImageCreateInfo-usage-00964](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap12.html#VUID-VkImageCreateInfo-usage-00964)
> [VUID-VkImageCreateInfo-Format-02536](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap12.html#VUID-VkImageCreateInfo-Format-02536)

I think enforcing a limit of `LIMIT_MAX_TEXTURE_SIZE_*` would forbid some otherwise-valid usecases, and possibly miss other restrictions.

Reply:
Its probably worth investigating what kind of limitation we would be putting on people if we just used ``maxImageDimension``. My guess is that in most cases all image types have the same size limitation. 

However, if there is a big difference I guess we can query max size on a case-by-case basis using [``vkGetPhysicalDeviceImageFormatProperties()``](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap47.html#vkGetPhysicalDeviceImageFormatProperties)